services/core/shared/core/blocks.js

```diff
@@ -0,0 +1,207 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { Logger } = require('lisk-service-framework');
+
+const logger = Logger();
+
+const pouchdb = require('../pouchdb');
+const coreApi = require('./compat');
+const config = require('../../config');
+
+const indexList = [
+	'id',
+	'generatorPublicKey',
+	'generatorAddress',
+	'generatorUsername',
+	'height',
+	'numberOfTransactions',
+	'previousBlockId',
+	'unixTimestamp',
+	'totalAmount',
+	'totalFee',
+	'isFinal',
+	['height', 'isFinal'],
+	['generatorPublicKey', 'numberOfTransactions'],
+	['generatorAddress', 'numberOfTransactions'],
+	['generatorUsername', 'numberOfTransactions'],
+	['generatorPublicKey', 'totalAmount'],
+	['generatorAddress', 'totalAmount'],
+	['generatorUsername', 'totalAmount'],
+	['generatorPublicKey', 'unixTimestamp'],
+	['generatorAddress', 'unixTimestamp'],
+	['generatorUsername', 'unixTimestamp'],
+];
```

Comment:
Consider moving the `indexList` to the config file.

Reply:
The `indexList` will be moved to the config.